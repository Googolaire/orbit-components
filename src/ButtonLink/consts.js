// @flow
export const TYPES = {
  PRIMARY: "primary",
  SECONDARY: "secondary",
};

export const TOKENS = {
  background: "background",
  backgroundHover: "backgroundHover",
  backgroundActive: "backgroundActive",
  backgroundFocus: "backgroundFocus",
  foreground: "foreground",
  foregroundHover: "foregroundHover",
  foregroundActive: "foregroundActive",
  foregroundFocus: "foregroundFocus",
  marginRightIcon: "marginRightIcon",
};
