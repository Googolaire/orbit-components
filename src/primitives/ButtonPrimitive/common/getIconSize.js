// @flow
import { ICON_SIZES } from "../../../Icon/consts";
import type { GetIconSize } from "./getIconSize";

const getIconSize: GetIconSize = size =>
  size === ICON_SIZES.SMALL ? ICON_SIZES.SMALL : ICON_SIZES.MEDIUM;

export default getIconSize;
