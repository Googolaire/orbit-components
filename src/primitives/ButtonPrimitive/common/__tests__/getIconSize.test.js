// @flow
import getIconSize from "../getIconSize";
import { SIZE_OPTIONS } from "../consts";

describe("ButtonPrimitive: getIconSize function", () => {
  it("should match snapshot", () => {
    expect(getIconSize(SIZE_OPTIONS.SMALL)).toMatchSnapshot();
    expect(getIconSize(SIZE_OPTIONS.LARGE)).toMatchSnapshot();
    expect(getIconSize(SIZE_OPTIONS.NORMAL)).toMatchSnapshot();
  });
});
