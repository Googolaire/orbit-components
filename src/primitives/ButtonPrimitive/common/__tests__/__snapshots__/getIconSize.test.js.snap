// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimitive: getIconSize function should match snapshot 1`] = `"small"`;

exports[`ButtonPrimitive: getIconSize function should match snapshot 2`] = `"medium"`;

exports[`ButtonPrimitive: getIconSize function should match snapshot 3`] = `"medium"`;
